test/integration/demux/signals_test.rb

```diff
@@ -33,6 +33,35 @@ class SignalsTest < ActionDispatch::IntegrationTest
       assert_requested(reporting_post)
     end
 
+    test "sending a signal with context" do
+      lesson = lessons(:first_lesson)
+
+      slack_post = stub_request(:post, demux_apps(:slack).signal_url)
+        .with(
+          body: hash_including(
+            action: "destroyed",
+            company_id: 1,
+            lesson: {
+              id: lesson.id,
+              name: lesson.name,
+              public: lesson.public
+            }
+          ),
+          headers: {
+            "Content-Type" => "application/json",
+            "User-Agent" => "Demux",
+            "X-Demux-Signal" => "lesson",
+            "X-Demux-Signature" => /\w.+/
+          }
+        )
+      reporting_post = stub_request(:post, demux_apps(:reporting).signal_url)
+
+      LessonSignal.new(lesson, account_id: lesson.company_id).destroyed
```

Comment:
Could you explain how the context is being sent here again? I'm pretty sure it is based on the automatic injection here: https://github.com/lessonly/demux/pull/15/files#diff-ea773719bcb501c54e004f9deb66238aR29 but wanted to confirm it. If so do we also need another test case for when specific arguments are used with a signal action? Example: https://github.com/lessonly/demux/pull/15/files#diff-04c6e90faac2675aa89e2176d2eec7d8R199

Reply:
For sure!

context is ultimately passed in from the action. So this action in the test dummy is:

```Ruby
def destroyed
    send :destroyed, context: destroyed_context
end
```

The context that is being passed into send is just calling a private method (it could have just been inline, the method is just organizational here):

```Ruby
  def destroyed_context
    {
      lesson: {
        id: lesson.id,
        name: lesson.name,
        public: lesson.public
      }
    }
  end
```

This method is just constructing a context based on the `lesson` object that was passed into the signal when initializing it. Since the state of that lesson is perishable (it's destroyed and not in the DB for later retrieval) that is why we are using context here to pull attributes off of the object passed in in order to build a context that we pass to the call to `send`.

I think we're fine without another example right now, because both methods are following the same path. They both end up passing context into `send`, in this case by pulling attributes off of the lesson object passed in and in the other case by accepting argument so the call to the action method. But in both cases, they are getting passed to the `context` argument of `send.